experimental/ast/expr_literal.go

```diff
@@ -19,6 +19,10 @@ import (
 )
 
 // ExprLiteral is an expression corresponding to a string or number literal.
+//
+// # Grammar
+//
+//	ExprLiteral := Number | String
```

Comment:
Consider using snake_case for lexical elements, to make it visually clear that they are different (and not defined herein).
```suggestion
//	ExprLiteral := number | string
```

Reply:
How about `token.Number` and `token.String`?

Reply:
Okay, that looks good to me. But you do also need to update the relevant comment in `doc.go` (~line 49).